src/equations.ml

```diff
@@ -605,7 +605,15 @@ let subst_rec_split env evd f comp comprecarg prob s split =
 		   if mem_assoc (out_name (get_name d)) s then acc
 		   else (mapping_constr subst c) :: acc
 		 else (mapping_constr subst c) :: acc) 0 [] args 
-	   in (mkEvar (ev', [||]), rev args'), !refarg
+           in
+           let secvars =
+             let named_context = Environ.named_context env in
+               List.map (fun decl ->
+                 let id = Context.Named.Declaration.get_id decl in
+                 Constr.mkVar id) named_context
+           in
+           let secvars = Array.of_list secvars in
+	    (mkEvar (ev', secvars), rev args'), !refarg
```

Comment:
If this is the correct way and it is not present in the Coq codebase (which surprised me a bit), then sure.